Lib/pathlib.py

```diff
@@ -430,11 +423,7 @@ def link_to(self, target):
     replace = os.replace
 
     if nt:
-        if supports_symlinks:
-            symlink = os.symlink
-        else:
-            def symlink(a, b, target_is_directory):
-                raise NotImplementedError("symlink() not available on this system")
+        symlink = os.symlink
```

Comment:
Why is it necessary to special case `os.symlink`? On non-Windows platforms, `os.symlink` ignores the `target_is_directory` parameter.

Reply:
cc @pitrou since this is done so from the initial implementation. I assume this is a leftover from older Python versions where `target_is_directory` was not accepted on POSIX?